### PR TITLE
resolve several trivial issues found by Coverity

### PR DESCRIPTION
**Pull request by chipitsine**
Changes proposed in this pull request:
 - improving coverity reports, no real impact or changes
 - 
 - 

Your great patch is much appreciated. We are considering to apply your patch into the SoftEther VPN main tree.

SoftEther VPN Patch Acceptance Policy:
http://www.softether.org/5-download/src/9.patch

You have two options which are described on the above policy.
Could you please choose either option 1 or 2, and specify it clearly on the reply?

- one

PRELIMINARY DECLARATION FOR FUTURE SWITCH TO A NON-GPL LICENSE

I hereby agree in advance that my work will be licensed automatically under the Apache License or a similar BSD/MIT-like open-source license in case the SoftEther VPN Project adopts such a license in future.

### CODE REVIEW

**Comment by davidebeatrici** [comment]
`g_zero` is now unused:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/30a554c1902407a7ad3118833c77604796473b17/src/Mayaqua/Kernel.c#L174

**Comment by chipitsine** [comment]
I'll remove it.

what about "DoNothing", is it safe to remove it ? I do not have idea why we need such function

**Comment by davidebeatrici** [comment]
It was used to invert the if statement.

File: src/Cedar/Interop_OpenVPN.c [hunk 1]
```diff
@@ -343,13 +343,8 @@ void OvsProceccRecvPacket(OPENVPN_SERVER *s, UDPPACKET *p, UINT protocol)
...
 									size -= sizeof(UINT);
 
-									if (size >= sizeof(ping_signature) &&
-										Cmp(data, ping_signature, sizeof(ping_signature)) == 0)
-									{
-										// Ignore since a ping packet has been received
-										DoNothing();
-									}
-									else
+									if (size < sizeof(ping_signature) ||
```
Comment (davidebeatrici) [inline src/Cedar/Interop_OpenVPN.c]: This is wrong, as every packet larger than `ping_signature` is ignored.

It should be:
```c
if (! (size >= sizeof(ping_signature) && Cmp(data, ping_signature, sizeof(ping_signature)) == 0))
```

File: src/Cedar/Interop_OpenVPN.c [hunk 2]
```diff
@@ -343,13 +343,8 @@ void OvsProceccRecvPacket(OPENVPN_SERVER *s, UDPPACKET *p, UINT protocol)
...
+										Cmp(data, ping_signature, sizeof(ping_signature)) != 0)
```
Comment (davidebeatrici) [inline src/Cedar/Interop_OpenVPN.c]: This is wrong, as every packet larger than `ping_signature` is ignored.

It should be:
```c
if (! (size >= sizeof(ping_signature) && Cmp(data, ping_signature, sizeof(ping_signature)) == 0))
```

Comment (davidebeatrici) [reply src/Cedar/Interop_OpenVPN.c]: Or, for better readability:
```c
const bool isPingPacket = size >= sizeof(ping_signature) && Cmp(data, ping_signature, sizeof(ping_signature)) == 0;
if (!isPingPacket)
```

Comment (chipitsine) [reply src/Cedar/Interop_OpenVPN.c]: I reverted "if (A && B)" to "if( not A || not B)" (at least I meant that)

Comment (davidebeatrici) [reply src/Cedar/Interop_OpenVPN.c]: Oh, sorry, I didn't notice you changed `== 0`  to `!= 0`.

**Pull request merged**